### PR TITLE
CLEANUP: environment-check: dependent executables error logic refactored

### PR DESCRIPTION
**Pull request by amelhusic**
Since only one error is captured, there is no need for `make([]error, 2)` and index logic.
Instead, last error is captured and returned.

Additionally, `sync.WaitGroup` exists, but there was no goroutines spawned.
So, added `go` keyword to spawn goroutines.

### CODE REVIEW

File: haproxy/haproxy_cmd/run.go [hunk 1]
```diff
@@ -118,31 +118,29 @@ func execAndCapture(path string, re *regexp.Regexp) (string, error) {
...
+			err = e
 		} else if strings.Compare(v, "1.2") < 0 {
-			errors[idx] = fmt.Errorf("%s version must be > 1.2,, but is: %s", path, v)
+			err = fmt.Errorf("%s version must be > 1.2, but is: %s", path, v)
 		}
-		wg.Done()
 	}
-	func() {
-		ensureVersion(haproxyBin, "^HA-Proxy version ([0-9]\\.[0-9]\\.[0-9])", "2.0", 0)
+	go func() {
```
Comment (ShimmerGlass) [inline haproxy/haproxy_cmd/run.go:134]: Nitpick: this can be just `go ensureVersion(...)`

**Comment by ShimmerGlass** [comment]
I just left a small nitpick comment since the goal is cleaning up. LGTM

File: haproxy/haproxy_cmd/run.go (hunk 1 above)
Comment (amelhusic) [reply haproxy/haproxy_cmd/run.go]: Sure, thanks :) Updated

**Comment by ShimmerGlass** [comment]
I just noticed that the original code for this is wrong. The lambda hardcodes the version number instead of using the argument and the strings.Compare won't work for versions like `0.1.10 > 0.1.2`.
This is not the scope of this PR tho, I can fix it after if you want.
Overwise LGTM :)

**Comment by amelhusic** [comment]
I found there is a library for comparing versions: https://github.com/hashicorp/go-version. If you wish, I can open another PR to address this issue?

**Comment by ShimmerGlass** [comment]
Sure :)

**Pull request merged**